### PR TITLE
Push serialization of history branch from history manager down to history store

### PR DESCRIPTION
**Pull request by norberthu**
**What changed?**

**Why?**

**How did you test it?**

**Potential risks**

**Is hotfix candidate?**

### CODE REVIEW

File: common/persistence/history_manager.go [hunk 1]
```diff
@@ -532,10 +532,17 @@ func (m *executionManagerImpl) NewHistoryBranch(
 	request *NewHistoryBranchRequest,
 ) (*NewHistoryBranchResponse, error) {
 
-	resp := &NewHistoryBranchResponse{}
-	var err error
-	resp.BranchToken, err = NewHistoryBranchToken(request.TreeId)
-	return resp, err
+	req := &NewHistoryBranchRequest{
```
Comment (yycptt) [inline common/persistence/history_manager.go:535]: nit: is this the same as the request parameter? resp is also the same?

**Review by yycptt — COMMENTED** [review]
Shall we exclude `NewHistoryBranch` from persistence rate limiting client. (maybe also retryable client) as it's not actually hitting DB or performing rpc call?

File: common/persistence/history_manager.go (hunk 1 above)
Comment (norberthu) [reply common/persistence/history_manager.go:535]: good catch! For the longest time I actually had a parallel `Internal-` prefixed struct and decided last minute that was unnecessary for this particular function. I'll clean this up

**Comment by norberthu** [comment]
Regarding rate limiting and retryable for `NewHistoryBranch`, while I don't have any immediate plans, there's a non-zero chance this might be necessary in the near future.

**Review by yux0 — APPROVED** [review]

**Pull request merged**